src/qt/rpcconsole.cpp

```diff
@@ -464,6 +446,101 @@ void RPCExecutor::request(const QString &command, const WalletModel* wallet_mode
+ *
+ * @param[in]    command  Command line to execute
+ * @param[in]    wallet_model  Wallet model to use
+ * @return  true if command was handled by this method (even on errors), false otherwise
+ *
+ */
+
+bool RPCExecutor::executeConsoleOnlyCommand(const std::string& command, const WalletModel* wallet_model)
+{
+    std::string result;
+    std::string address;
+    std::string executableCommand;
+    std::vector<std::string> parsedCommand{parseHelper(command)};
+
+    // Catch the console-only generate command with 2 or less parameters before RPC call is executed .
+    if (!parsedCommand.empty() && parsedCommand[0] == "generate" && parsedCommand.size() <= 3) {
+        // Default number of blocks is 1 if missing
+        const std::string nblocks{parsedCommand.size() > 1 ? parsedCommand[1] : "1"};
+        // An empty maxtries parameters will be modified by generatetoaddress' default value
+        const std::string maxtries{parsedCommand.size() > 2 ? parsedCommand[2] : ""};
+
+        // Generate address
+        if (!RPCConsole::RPCExecuteCommandLine(m_node, address, "getnewaddress\n", nullptr, wallet_model)) {
+            Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Error: could not generate new address"));
+        } else {
+            executableCommand = "generatetoaddress " + nblocks + " \"" + address + "\" " + maxtries;
+            if (!RPCConsole::RPCExecuteCommandLine(m_node, result, "generatetoaddress " + nblocks + " " + address + " " + maxtries + "\n", nullptr, wallet_model)) {
+                    Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Error: could not generate blocks"));
+            } else {
+                    std::string answer = "{\n  \"address\": \"" + address + "\",\n  \"blocks\": " + result + "\n}";
+                    Q_EMIT reply(RPCConsole::CMD_REPLY, QString::fromStdString("\n" + answer + "\n\n"));
+            }
+        }
+        return true;
+    }
+    // Catch the console-only "help generate" command when requested, or when "generate is called with wrong parameters.
+    if ((parsedCommand.size() >= 2 && parsedCommand[0] == "help" && parsedCommand[1] == "generate") || (parsedCommand.size() > 3 && parsedCommand[0] == "generate")) {
+        Q_EMIT reply(RPCConsole::CMD_REPLY, QString(("\n"
+                                                     "Generate blocks, equivalent to RPC getnewaddress followed by RPC generatetoaddress.\n"
+                                                     "Optional integer arguments are number of blocks to generate and maximum iterations to try.\n"
+                                                     "Equivalent to RPC generatetoaddress nblocks and maxtries arguments.\n"
+                                                     "   example:    generate\n"
+                                                     "   example:    generate 4\n"
+                                                     "   example:    generate 3 6000\n\n")));
```

Comment:
Formatting-only suggestion (personal preference to avoid long lines), and the argument to `QString()` had extra parentheses. Could make a similar change below.
```suggestion
        if ((parsedCommand.size() > 1 && parsedCommand[0] == "help" && parsedCommand[1] == "generate") ||
            (parsedCommand.size() > 3 && parsedCommand[0] == "generate")) {
                auto message{
                    "\n"
                    "Generate blocks, equivalent to RPC getnewaddress followed by RPC generatetoaddress.\n"
                    "Optional integer arguments are number of blocks to generate and maximum iterations to try.\n"
                    "Equivalent to RPC generatetoaddress nblocks and maxtries arguments.\n"
                    "   example:    generate\n"
                    "   example:    generate 4\n"
                    "   example:    generate 3 6000\n\n"};
                Q_EMIT reply(RPCConsole::CMD_REPLY, QString(message));
```

Reply:
Removed extra parenthesis, and I agree with extra long lines, but as I said before, indentation was introduced after running the script suggested by @ jarolrod and are apparently ok according to clang format standard.